enclave/core/sgx/init.c

```diff
@@ -41,33 +41,17 @@ static void _check_memory_boundaries(void)
 }
 
 #ifdef OE_WITH_EXPERIMENTAL_EEID
-extern volatile const oe_sgx_enclave_properties_t oe_enclave_properties_sgx;
-extern oe_eeid_t* oe_eeid;
-
-static int _is_eeid_base_image(
-    const volatile oe_sgx_enclave_properties_t* properties)
-{
-    return properties->header.size_settings.num_heap_pages == 0 &&
-           properties->header.size_settings.num_stack_pages == 0 &&
-           properties->header.size_settings.num_tcs == 1;
-}
-
-static oe_result_t _eeid_patch_memory()
+static void _find_eeid()
 {
-    oe_result_t r = OE_OK;
-
-    if (_is_eeid_base_image(&oe_enclave_properties_sgx))
-    {
-        uint8_t* enclave_base = (uint8_t*)__oe_get_enclave_base();
-        uint8_t* heap_base = (uint8_t*)__oe_get_heap_base();
-        oe_eeid_marker_t* marker = (oe_eeid_marker_t*)heap_base;
-        oe_eeid = (oe_eeid_t*)(enclave_base + marker->offset);
-
-        // Wipe the marker page
-        memset(heap_base, 0, OE_PAGE_SIZE);
-    }
+    /* Get the guard/EEID page */
+    const oe_eeid_page_t* eeid_page = (oe_eeid_page_t*)__oe_get_eeid_page();
+    if (!oe_is_within_enclave(eeid_page, OE_PAGE_SIZE))
+        oe_abort();
 
-    return r;
+    /* Without EEID, eeid_page page is all zero (including the version). */
+    if (eeid_page->eeid.version != 0 &&
+        eeid_page->eeid.version != OE_EEID_VERSION)
+        oe_abort();
```

Comment:
Why does this abort if the page is not populated? From the comment in loadelf.c `_patch()`, it sounds like its valid for this page to exist but not be populated with eeid info if not used.

Reply:
That's right, in that case the version is 0, because the whole page is zero. This check aborts if the version doesn't match 0 or the currently supported version (e.g. garbage).

Reply:
This has been refactored. There is a new global `_eeid_rva` now that indicates whether (and where) the EEID is.